### DISCUSSION TITLE
timeout trying View() a large dataframe

### DISCUSSION THREAD

**Discussion by brndngrhm** [posit-dev/positron#4424] [Q&A]
I have a 2 million row dataframe and when i run `View(my_data)` in the console the data explorer doesn't open but also there is no error, just nothing happens. Also trying to expand the column names dropdown results this timeout error ![image](https://github.com/user-attachments/assets/5e73816e-3c8c-4cf9-b823-8dea19101d48)

**Comment by juliasilge** [comment]
Those errors are coming from the Variables pane (not the Data Explorer) and I believe this is the same as #2223.

Did you see any errors that mentioned the Data Explorer itself?

**Comment by brndngrhm** [comment]
thanks - also found a similar issue in #3628. 

No, no errors mentioning the Data Explorer. But i can't get the data explorer to even open using either `View(my_data)` or by clicking on the grid icon in the variables pane. There is no error, just nothing happens. I re-started the console a few times but no luck.

**Reply by juliasilge** [reply]
Hmmmm, interesting. Can you create a reproducible example that demonstrates the problem you are observing?

For example, if I do this (5 million rows):

```r
foo <- nycflights13::flights
foo <- rbind(foo, foo)
foo <- rbind(foo, foo)
foo <- rbind(foo, foo)
foo <- rbind(foo, foo)
foo <- cbind(foo, foo)
```

Then I _can_ successfully do `View(foo)` and the Data Explorer works nicely (but I do see the errors in the Variables pane as described elsewhere).

**Comment by brndngrhm** [comment]
After i completely closed out of Positron and opened it back up the `nycflights13` example worked, and I could open the data explorer with my real data. Thought i could just restart the console but that wasn't enough.

I tried to recreate my "nothing happens" scenario and couldn't, but i did find that repeatedly clicking on the variable drop down arrows (like 10-15 times in rapid succession) causes a huge about of CPU to get used and R/positron seemingly crashed and i got the errors below:

![image](https://github.com/user-attachments/assets/58c1a858-cac1-421c-960f-cbded2f292c9)

![image](https://github.com/user-attachments/assets/f1b7303d-1a6b-48c0-a7c4-7fc349935a2b)

**Reply by juliasilge** [reply]
OK yep, my current hypothesis is that there isn't anything wrong with how the Data Explorer handles these large dataframes, but that the Variables pane gets super busted. 😆 Thank you for the report!
